src/components/shadow.js

```diff
@@ -0,0 +1,43 @@
+var component = require('../core/component');
+var THREE = require('../lib/three');
+var bind = require('../utils/bind');
+var registerComponent = component.registerComponent;
+
+/**
+ * Shadow component.
+ *
+ * When applied to an entity, that entity's geometry and any descendants will cast or receive
+ * shadows as specified by the `cast` and `receive` properties.
+ */
+module.exports.Component = registerComponent('shadow', {
+  schema: {
+    cast: {default: true},
+    receive: {default: true}
+  },
+
+  init: function () {
+    this.onMeshChanged = bind(this.update, this);
+    this.el.addEventListener('model-loaded', this.onMeshChanged);
+  },
+
+  update: function () {
```

Comment:
What happens if we add entities dynamically to the scene? Should we listen for the `loaded` event?